prober/tcp_test.go

```diff
@@ -193,6 +201,158 @@ func TestTCPConnectionWithTLS(t *testing.T) {
 	checkRegistryResults(expectedResults, mfs, t)
 }
 
+func TestTCPConnectionWithTLSAndVerifiedCertificateChain(t *testing.T) {
+	if os.Getenv("TRAVIS") == "true" {
+		t.Skip("skipping; travisci is failing on ipv6 dns requests")
+	}
+
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatalf("Error listening on socket: %s", err)
+	}
+	defer ln.Close()
+	_, listenPort, _ := net.SplitHostPort(ln.Addr().String())
+
+	testCTX, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	rootPrivatekey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		panic(fmt.Sprintf("Error creating rsa key: %s", err))
+	}
+
+	// Prepare certificates to simulate a situation where
+	// the root certificate in the chain sent by the server
+	// expired but a verified certificate chain can be found
+	// because a new one has installed on the local machine.
```

Comment:
"one" is ambiguous here. Root?


This also feels like we're testing Go's root handling rather than testing our own code. Having two chains where one is expired would seem like a better test.

Reply:
Right. I made a change to do that in https://github.com/prometheus/blackbox_exporter/pull/636/commits/d45159b5485d39fd06cdad71f2c9fc56c630a6ff